src/resolve/resolved-bus.c

```diff
@@ -523,6 +524,9 @@ static int bus_method_resolve_hostname(sd_bus_message *message, void *userdata,
 
         q->bus_request = sd_bus_message_ref(message);
         q->request_family = family;
+        q->request_name = hostname;
+        if (!q->request_name)
```

Comment:
this is a strange check, there's no memory allocation here?

Reply:
Yes, this is unnecessary and has been removed now.